### ISSUE TITLE
Serious regression: /**< comments cause functions to be silently ignored

### ISSUE THREAD

**Issue by mosra** [doxygen/doxygen#7472]
Minimal repro case. I remember this working with fresh 1.8.16, but then it got broken on the way to 1.8.17. Not sure which commit is to blame (and I didn't have time to bisect), the changelog didn't list anything suspicious:

```cpp
/** @file
 * @brief A file
 */

/** A thing. Only the int and double overloads show, the float one doesn't */
void foo();

/** Shows up */
void foo(int a) {}

void foo(float a) {} /**< Doesn't show up */

void foo(double a); /**< This shows up, tho */
```

What's extra bad is that the `float` overload just gets ignored, **with no warning or anything**. The only chance of discovering this is when some other code tries to link to that particular function, which is not always the case.

Originally I thought this affects just functions marked with `/**< @overload */` (because that's where I discovered this), but seems to be for anything where `/**<` follows a function block.

**Comment by albert-github** [comment]
I've done a "bisect" and with the version:

> Commit: 4e21d19cf63b17aadb62905535262deb208a0c73 [4e21d19]
> Date: Saturday, September 28, 2019 12:08:11 PM
> Merge pull request #7282 from albert-github/feature/issue_7274
> 
> issue #7274 Subpages no longer possible under mainpage

the output is still conform the 1.8.16 output.
With the, next, commit in master:

> Commit: d2b8ae16cb99621827f5dd860222c27e629f5a7f [d2b8ae1]
> Date: Monday, October 7, 2019 9:02:04 PM
> Merge branch 'std'

the output of the "float" function is gone.

**Comment by mosra** [comment]
Oh, a merge commit, that's unfortunate. What caught my attention is this part, especially the TODO -- could it be related?

```cpp
    /*! @name add entry as a child and pass ownership.
     *  @note This makes the entry passed invalid! (TODO: tclscanner.l still has use after move!)
     *  @{
     */
```

**Comment by albert-github** [comment]
I don't think so as the tclscanner is not called, but I'm not 100% sure whether or not it has side effects (still an old call? I didn't see it at first glance).

**Comment by doxygen** [comment]
I'm expecting the issue is in scanner.l, but it is not directly obvious so we have to look at it in more detail. My suspicion is that the old implementation depended on either the `current` or `previous` object containing the documentation and this data is now being cleared before it is used.

**Comment by doxygen** [comment]
@mosra Please check if the proposed fix works for you

**Comment by mosra** [comment]
Yes it does! :) Thank you so much for the quick fix.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.18.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
